src/administration/Administration.Service/BusinessLogic/RegistrationBusinessLogic.cs

```diff
@@ -181,7 +185,42 @@ public Task<Pagination.Response<CompanyApplicationDetails>> GetCompanyApplicatio
                     .AsAsyncEnumerable()));
     }
 
-    public Task<Pagination.Response<CompanyApplicationWithCompanyUserDetails>> GetAllCompanyApplicationsDetailsAsync(int page, int size, string? companyName = null)
+    public Task<Pagination.Response<CompanyDetailsOspOnboarding>> GetOspCompanyDetailsAsync(int page, int size, CompanyApplicationStatusFilter? companyApplicationStatusFilter, string? companyName)
+    {
+        if (!string.IsNullOrEmpty(companyName) && !Company.IsMatch(companyName))
+        {
+            throw new ControllerArgumentException("CompanyName length must be 3-40 characters and *+=#%\\s not used as one of the first three characters in the company name", nameof(companyName));
+        }
+        var applications = _portalRepositories.GetInstance<IApplicationRepository>()
+            .GetExternalCompanyApplicationsFilteredQuery(_identityData.CompanyId,
+                companyName?.Length >= 3 ? companyName : null,
+                GetCompanyApplicationStatusIds(companyApplicationStatusFilter));
+
+        return Pagination.CreateResponseAsync(
+            page,
+            size,
+            _settings.ApplicationsMaxPageSize,
+            (skip, take) => new Pagination.AsyncSource<CompanyDetailsOspOnboarding>(
+                applications.CountAsync(),
+                applications
+                    .AsSplitQuery()
+                    .OrderByDescending(application => application.DateCreated)
+                    .Skip(skip)
+                    .Take(take)
+                    .Select(application => new CompanyDetailsOspOnboarding(
+                        application.CompanyId,
+                        application.Id,
+                        application.ApplicationStatusId,
+                        application.DateCreated,
+                        application.DateLastChanged,
+                        application.Company!.Name,
+                        application.Company.CompanyAssignedRoles.Select(companyAssignedRoles => companyAssignedRoles.CompanyRoleId),
+                        application.Company.IdentityProviders.Select(x => new IdentityProvidersDetails(x.Id, x.IamIdentityProvider!.IamIdpAlias)),
+                        application.Company.BusinessPartnerNumber,
+                        application.Company.Identities.Where(x => x.CompanyUser!.Identity!.UserStatusId != UserStatusId.DELETED).Count()))
```

Comment:
This can be changed to

```suggestion
                        application.Company.Identities.Count(x => x.CompanyUser!.Identity!.UserStatusId != UserStatusId.DELETED)))
```